File: date_today/golang_date_today/date_today.go
package main

import "os"
import "fmt"
import "time"

func date_today() {
  local := time.Now()
  year := local.Year()
  month := local.Month()
  day := local.Day()
  hour := local.Hour()
  min := local.Minute()
  sec := local.Second()
  fmt.Printf("local=<%d-%d-%d %d:%d:%d>\n", year, month, day, hour, min, sec)
  fmt.Println("local=" + local.String())
}

func date_utc() {
  utc := time.Now().UTC()
  year := utc.Year()
  month := utc.Month()
  day := utc.Day()
  hour := utc.Hour()
  min := utc.Minute()
  sec := utc.Second()
  fmt.Printf("utc=<%d-%d-%d %d:%d:%d>\n", year, month, day, hour, min, sec)
  fmt.Println("utc=" + utc.String())
}

func main() {
  args := os.Args
  argc := len(args)

  if argc != 1 {
    fmt.Println(args[0] + " <noargs>")
    os.Exit(1)
  }

  date_today();
  date_utc();
}

File: date_today/kotlin_date_today/DateToday.kt
package DateToday

import java.time.*
//import java.time.LocalDate

//comment
class DateToday {
  fun date_utc() {
    val utc: ZonedDateTime = ZonedDateTime.now(ZoneOffset.UTC)
    val year = utc.getYear()
    val month = utc.getMonthValue()
    val day = utc.getDayOfMonth()
    val hour = utc.getHour()
    val min = utc.getMinute()
    val sec = utc.getSecond()
    println("utc=<" + year + "-" + month + "-" + day + " " + hour + ":" + min + ":" + sec + ">")
  }

  fun date_today() {
    val local: LocalDateTime = LocalDateTime.now()
    val year = local.getYear()
    val month = local.getMonthValue()
    val day = local.getDayOfMonth()
    val hour = local.getHour()
    val min = local.getMinute()
    val sec = local.getSecond()
    println("local=<" + year + "-" + month + "-" + day + " " + hour + ":" + min + ":" + sec + ">")
  }
}

fun main(args : Array<String>) {
  if( args.size != 0 ) {
    println("Usage: java -jar DateToday.jar <noargs>")
    System.exit(1)
  }
  val date_today = DateToday()
  date_today.date_today()
  date_today.date_utc()
}

